### PR TITLE
logger class converted to a singleton

### PR DESCRIPTION
**Pull request by 3lv27**

### CODE REVIEW

File: src/logger.js [hunk 1]
```diff
@@ -1,8 +1,12 @@
 // A simple logging function
 export default class Logger {
   constructor(system, verbose) {
-    this.system = system;
-    this.verbose = verbose;
+    if(!Logger.instance) {
```
Comment (gmuraru) [inline src/logger.js]: CS: Only a matter of preferences, but I have seen that the files are written with a "space" character between the ```if clause``` and the ```open parenthesis```.

Comment (cereallarceny) [reply src/logger.js]: This should be caught by using Prettier.

**Comment by cereallarceny** [comment]
@3lv27 You will also need to update all places where we pass the `Logger` class.  Please also run your test suite to ensure that if tests are failing, it's not because of you.  I believe the tests on the execution branch fail, but they shouldn't fail because of your PR if it works correctly.

**Comment by 3lv27** [comment]
@cereallarceny There's no need to update all places where we pass the Logger class since the singleton pattern is happening in the constructor itself, whenever we'll call the class it is going to return the instance already created if there's any. Regarding the test, everything is working fine in local for me

**Comment by cereallarceny** [comment]
I understand, but it will provide for cleaner code.  Now that it's no longer needed to be passed as an argument, let's not do that.  Please change your PR to instantiate a `new Logger` in each file where it's called and eliminate references to the logger as a function argument.  After that, I will accept this PR.

**Pull request closed without merging**